drivers/UARTSerial.cpp

```diff
@@ -407,6 +419,17 @@ int UARTSerial::enable_output(bool enabled)
     return 0;
 }
 
+void UARTSerial::init() {
+    for (size_t i = 0; i < sizeof _irq / sizeof _irq[0]; i++) {
+        _irq[i] = NULL;
+    }
+
+    serial_init(&_serial, _serial.pin_tx, _serial.pin_rx);
+    serial_baud(&_serial, _baud);
+    serial_irq_handler(&_serial, SerialBase::_irq_handler, (uint32_t)this);
+    enable_rx_irq();
```

Comment:
This enable is not correct. You only want to enable interrupts based on buffer state and enable status. As you're calling this for TX inside `if (!rx_enabled)`, you know you don't want the RX interrupt enabled. And vice versa.